src/components/menu/popoverMenu.tsx

```diff
@@ -0,0 +1,107 @@
+import './popoverMenu.css'
+
+import MoreVertIcon from '@mui/icons-material/MoreVert'
+import { useMediaQuery } from '@mui/material'
+import Drawer from '@mui/material/Drawer'
+import IconButton from '@mui/material/IconButton'
+import ListItemIcon from '@mui/material/ListItemIcon'
+import ListItemText from '@mui/material/ListItemText'
+import Menu from '@mui/material/Menu'
+import MenuItem from '@mui/material/MenuItem'
+import { useTheme } from '@mui/material/styles'
+import Typography from '@mui/material/Typography'
+import type { ReactNode } from 'react'
+import React, { useState } from 'react'
+
+export interface PopoverMenuItem {
+  label: string
+  onClick: (event?: React.MouseEvent<HTMLElement>) => void
+  divider?: boolean
+  icon?: ReactNode
+  endDecorator?: ReactNode
+}
+
+export interface PopoverMenuProps {
+  menuItems: PopoverMenuItem[]
+  /** Aria-label to describe the button for assistive technology. */
+  ariaLabel: string
+  /** Provide your own customized icon. */
+  icon?: ReactNode
```

Comment:
should we limit the icon and endDecorator to be either of type Icon or SvgIcon ?

Reply:
Also, what would happen in cases where text is required in addition to or instead of icon? For example, consider a menu for `Settings` or `Payment` ?  If that is supported, I think the prop should be renamed more appropriately to clarify that. This should then be reflected in the storybook as well as tests. 
If not, please elaborate or document why this would not be supported.